src/pointcloud_to_laserscan_nodelet.cpp

```diff
@@ -192,12 +192,35 @@ namespace pointcloud_to_laserscan
       cloud_out = cloud_msg;
     }
 
+    bool flagFirstIteration=true;
+    sensor_msgs::PointCloud2ConstIterator<float> *iter_intensity;
     // Iterate through pointcloud
     for (sensor_msgs::PointCloud2ConstIterator<float>
               iter_x(*cloud_out, "x"), iter_y(*cloud_out, "y"), iter_z(*cloud_out, "z");
               iter_x != iter_x.end();
               ++iter_x, ++iter_y, ++iter_z)
     {
+      if(flagFirstIteration)
+      {
+        flagFirstIteration=false;
+        try
+        {
+          iter_intensity=new sensor_msgs::PointCloud2ConstIterator<float>(*cloud_out, "intensity");
```

Comment:
Is this a standard field for intensities in point clouds? It might be more 'explicit' to add a parameter to the node - intensity_field. When set, it can read that field in the pointcloud into the laserscan intensity, and error out otherwise.